### ISSUE TITLE
Intermittent E2E test failure (stateapp_test)

### ISSUE THREAD

**Issue by wcs1only** [dapr/dapr#2034]
## In what area(s)?

/area test-and-release

## What version of Dapr?

edge

## Expected Behavior

Test should pass.

## Actual Behavior

https://github.com/dapr/dapr/runs/1107312806

```
2020/09/13 02:41:20 Service ingress for stateapp is ready...
=== RUN   TestStateApp/Test_get_getbulk_save_delete_with_empty_request_response_for_single_app_and_single_hop
=== RUN   TestStateApp/Test_save_get_and_delete_a_single_item_for_single_app_and_single_hop
    stateapp_test.go:327: 
        	Error Trace:	stateapp_test.go:327
        	Error:      	Should be true
        	Test:       	TestStateApp/Test_save_get_and_delete_a_single_item_for_single_app_and_single_hop
=== RUN   TestStateApp/Test_save_get_and_delete_on_multiple_items_for_single_app_and_single_hop
    stateapp_test.go:327: 
        	Error Trace:	stateapp_test.go:327
        	Error:      	Should be true
        	Test:       	TestStateApp/Test_save_get_and_delete_on_multiple_items_for_single_app_and_single_hop
--- FAIL: TestStateApp (49.73s)
    --- PASS: TestStateApp/Test_get_getbulk_save_delete_with_empty_request_response_for_single_app_and_single_hop (0.72s)
    --- FAIL: TestStateApp/Test_save_get_and_delete_a_single_item_for_single_app_and_single_hop (1.03s)
    --- FAIL: TestStateApp/Test_save_get_and_delete_on_multiple_items_for_single_app_and_single_hop (0.71s)
=== RUN   TestStateTransactionApps
2020/09/13 02:41:44 Waiting until service ingress is ready for stateapp...
2020/09/13 02:41:44 Service ingress for stateapp is ready...
=== RUN   TestStateTransactionApps/Test_State_Transactions_using_HTTP_protocol
    stateapp_test.go:379: 
        	Error Trace:	stateapp_test.go:379
        	Error:      	Not equal: 
        	            	expected: "{\"data\":\"Hello World\"}"
        	            	actual  : ""
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-{"data":"Hello World"}
        	            	+
        	Test:       	TestStateTransactionApps/Test_State_Transactions_using_HTTP_protocol
```
## Steps to Reproduce the Problem

Intermittent E2E test failure.

**Comment by wcs1only** [comment]
Finally had time to triage this. It smells a lot like the component startup race in #1945, so this might already be fixed. For some reason, I'm having trouble getting the daprd container logs to confirm.
https://ms.portal.azure.com#@72f988bf-86f1-41af-91ab-2d7cd011db47/blade/Microsoft_Azure_Monitoring_Logs/LogsBlade/resourceId/%2Fsubscriptions%2F38875a89-0178-4f27-a141-dc6fc01f183d%2FresourceGroups%2Fdapre2e%2Fproviders%2FMicrosoft.ContainerService%2FmanagedClusters%2Fdapr-aks-e2e-07/source/LogsBlade.AnalyticsShareLinkToQuery/q/H4sIAAAAAAAAA12PTWsCMRCG7%252Fsr3psXcd3NfsSDJy2l4K2l9zGZ0chusiQRKfjjXQqtpbfh4XnfmSlLHFzKMMFncp4jhnBKmObB08hpIsMoyhKfjm%252F%252FLYlhBA0D8pmfeoLz38QM15Q5rlDsfnKHcMIdtzNHxi9822O7xUIJ1Y1VTdUKie3brtFitFRttzakuFO0UfrYWktGVFVvdK2IZU1H1fei2SyKO6YYLmwyPtzIrzx3U2a7%252FLtqifmGF5%252Fj13u4xvm5J3gA%252FPwb9wwBAAA%253D/timespan/2020-09-13T02%3A38%3A00.000Z%2F2020-09-13T02%3A45%3A00.000Z
```
2020/09/13 02:41:41 Posting state to http://localhost:3500/v1.0/state/statestore with 'null'
2020/09/13 02:41:41 Processing save request for 0 entries.
2020/09/13 02:41:41 Processing request for /test/http/get
2020/09/13 02:41:41 Result for get request for 0 states: []
2020/09/13 02:41:41 Processing get request for 0 states.
2020/09/13 02:41:41 indexHandler is called
2020/09/13 02:41:41 Processing request for /test/http/save
2020/09/13 02:41:42 Processing request for /test/http/delete
2020/09/13 02:41:42 Found state for key c401c204-ff64-49ee-8798-c277478bf65e: {"errorCode":"ERR_STATE_STORE_NOT_CONFIGURED","message":""}
2020/09/13 02:41:42 Result for get request for key c401c204-ff64-49ee-8798-c277478bf65e: &{}
2020/09/13 02:41:42 Result for get request for 1 states: [{c401c204-ff64-49ee-8798-c277478bf65e 0xc000021c80 }]
2020/09/13 02:41:42 Processing delete request for 0 states.
2020/09/13 02:41:42 Processing request for /test/http/get
2020/09/13 02:41:42 Processing get request for 1 states.
2020/09/13 02:41:42 Processing get request for c401c204-ff64-49ee-8798-c277478bf65e.
2020/09/13 02:41:42 Fetching state from http://localhost:3500/v1.0/state/statestore/c401c204-ff64-49ee-8798-c277478bf65e
2020/09/13 02:41:43 Fetching state from http://localhost:3500/v1.0/state/statestore/e9152294-d9af-4e32-8fee-633b09d0a5ef
2020/09/13 02:41:43 Processing get request for e9152294-d9af-4e32-8fee-633b09d0a5ef.
2020/09/13 02:41:43 Processing request for /test/http/get
2020/09/13 02:41:43 Processing get request for 5 states.
2020/09/13 02:41:43 Processing request for /test/http/save
2020/09/13 02:41:43 Processing save request for 1 entries.
2020/09/13 02:41:43 Posting state to http://localhost:3500/v1.0/state/statestore with '[{"key":"e9152294-d9af-4e32-8fee-633b09d0a5ef","value":{"data":"Value for entry #1 with key e9152294-d9af-4e32-8fee-633b09d0a5ef."}​},{"key":"d99f32a8-b656-4ee7-807c-7252e048c18d","value":{"data":"Value for entry #2 with key d99f32a8-b656-4ee7-807c-7252e048c18d."}​},{"key":"468f9e9f-5ae4-4507-9bbf-11a18d904064","value":{"data":"Value for entry #3 with key 468f9e9f-5ae4-4507-9bbf-11a18d904064."}​},{"key":"26cf2c0a-6735-4152-bda4-94b418d63500","value":{"data":"Value for entry #4 with key 26cf2c0a-6735-4152-bda4-94b418d63500."}​},{"key":"de617213-a678-41af-b632-814ad6375bc4","value":{"data":"Value for entry #5 with key de617213-a678-41af-b632-814ad6375bc4."}​}]'
2020/09/13 02:41:43 Found state for key 26cf2c0a-6735-4152-bda4-94b418d63500: {"errorCode":"ERR_STATE_STORE_NOT_CONFIGURED","message":""}
2020/09/13 02:41:43 Fetching state from http://localhost:3500/v1.0/state/statestore/26cf2c0a-6735-4152-bda4-94b418d63500
2020/09/13 02:41:43 Processing get request for 26cf2c0a-6735-4152-bda4-94b418d63500.
2020/09/13 02:41:43 Result for get request for key 468f9e9f-5ae4-4507-9bbf-11a18d904064: &{}
2020/09/13 02:41:43 Posting state to http://localhost:3500/v1.0/state/statestore with '[{"key":"c401c204-ff64-49ee-8798-c277478bf65e","value":{"data":"The best song ever is 'Highwayman' by 'The Highwaymen'."}​}]'
2020/09/13 02:41:43 Processing request for /test/http/get
2020/09/13 02:41:43 Processing get request for 1 states.
2020/09/13 02:41:43 Processing get request for c401c204-ff64-49ee-8798-c277478bf65e.
2020/09/13 02:41:43 Fetching state from http://localhost:3500/v1.0/state/statestore/c401c204-ff64-49ee-8798-c277478bf65e
2020/09/13 02:41:43 Found state for key c401c204-ff64-49ee-8798-c277478bf65e: {"errorCode":"ERR_STATE_STORE_NOT_CONFIGURED","message":""}
2020/09/13 02:41:43 Found state for key 468f9e9f-5ae4-4507-9bbf-11a18d904064: {"errorCode":"ERR_STATE_STORE_NOT_CONFIGURED","message":""}
2020/09/13 02:41:43 Result for get request for key c401c204-ff64-49ee-8798-c277478bf65e: &{}
2020/09/13 02:41:43 Result for get request for 1 states: [{c401c204-ff64-49ee-8798-c277478bf65e 0xc000021d90 }]
2020/09/13 02:41:43 Processing request for /test/http/get
2020/09/13 02:41:43 Processing get request for 5 states.
2020/09/13 02:41:43 Processing get request for e9152294-d9af-4e32-8fee-633b09d0a5ef.
2020/09/13 02:41:43 Result for get request for key 26cf2c0a-6735-4152-bda4-94b418d63500: &{}
2020/09/13 02:41:43 Processing get request for de617213-a678-41af-b632-814ad6375bc4.
2020/09/13 02:41:43 Result for get request for key e9152294-d9af-4e32-8fee-633b09d0a5ef: &{}
2020/09/13 02:41:43 Processing get request for d99f32a8-b656-4ee7-807c-7252e048c18d.
2020/09/13 02:41:43 Fetching state from http://localhost:3500/v1.0/state/statestore/d99f32a8-b656-4ee7-807c-7252e048c18d
2020/09/13 02:41:43 Found state for key d99f32a8-b656-4ee7-807c-7252e048c18d: {"errorCode":"ERR_STATE_STORE_NOT_CONFIGURED","message":""}
2020/09/13 02:41:43 Result for get request for key d99f32a8-b656-4ee7-807c-7252e048c18d: &{}
2020/09/13 02:41:43 Processing get request for 468f9e9f-5ae4-4507-9bbf-11a18d904064.
2020/09/13 02:41:43 Fetching state from http://localhost:3500/v1.0/state/statestore/468f9e9f-5ae4-4507-9bbf-11a18d904064
2020/09/13 02:41:43 Found state for key e9152294-d9af-4e32-8fee-633b09d0a5ef: {"errorCode":"ERR_STATE_STORE_NOT_CONFIGURED","message":""}
stderr
```

**Comment by wcs1only** [comment]
Yeah, I don't know why the daprd container logs are empty:

https://ms.portal.azure.com#@72f988bf-86f1-41af-91ab-2d7cd011db47/blade/Microsoft_Azure_Monitoring_Logs/LogsBlade/resourceId/%2Fsubscriptions%2F38875a89-0178-4f27-a141-dc6fc01f183d%2FresourceGroups%2Fdapre2e%2Fproviders%2FMicrosoft.ContainerService%2FmanagedClusters%2Fdapr-aks-e2e-07/source/LogsBlade.AnalyticsShareLinkToQuery/q/H4sIAAAAAAAAA12PQUsDMRSE7%252Fsr5tZLcRNfX8oeelIRoTfFe5p9aSO7yZKkFKE%252F3iBoxdvwMfPeTN9jH0qFS7HaECVjSseCpYloZymLdYKu7%252FEe5PLf5XOaYacJ9SQ3e0GI38RN51Il36F7%252BMnt0xFXXE6SBb%252Fw5RG7HVY8kHGkt6NmpZl5Mxg2MsiBaLN1bNUwesP6npQZiQ6ehcQbbZRSLeZp1V2x5PQhruItzPIs7batMq7%252FvlqjdXiKNX%252B%252BpnNu427gC%252BuMPz0MAQAA/timespan/P7D

But this is likely to be fixed by #2039
